storage/filestore/dir.go

```diff
@@ -81,32 +86,48 @@ func (dir *Dir) DeleteTemporary(ctx context.Context, file *os.File) (err error)
 	return errs.Combine(closeErr, os.Remove(file.Name()))
 }
 
-// blobToPath converts blob reference to a filepath in permanent storage
-func (dir *Dir) blobToPath(ref storage.BlobRef) (string, error) {
+// blobToBasePath converts a blob reference to a filepath in permanent storage. This may not be the
+// entire path; blobPathForFormatVersion() must also be used. This is a separate call because this
+// part of the filepath is constant, and blobPathForFormatVersion may need to be called multiple
+// times with different storage.FormatVersion values.
+func (dir *Dir) blobToBasePath(ref storage.BlobRef) (string, error) {
 	if !ref.IsValid() {
 		return "", storage.ErrInvalidBlobRef.New("")
 	}
 
 	namespace := pathEncoding.EncodeToString(ref.Namespace)
 	key := pathEncoding.EncodeToString(ref.Key)
 	if len(key) < 3 {
-		// ensure we always have at least
+		// ensure we always have enough characters to split [:2] and [2:]
 		key = "11" + key
 	}
 	return filepath.Join(dir.blobsdir(), namespace, key[:2], key[2:]), nil
 }
 
-// blobToTrashPath converts blob reference to a filepath in transient storage
-// the files in trash are deleted in an interval (in case the initial deletion didn't work for some reason)
+// blobPathForFormatVersion adjusts a bare blob path (as might have been generated by a call to
+// blobToBasePath()) to what it should be for the given storage format version.
+func blobPathForFormatVersion(path string, formatVersion storage.FormatVersion) string {
```

Comment:
Maybe `appendFormatVersionExt`?

Reply:
I'm not sure that this will always only append an extension. If future storage formats need to use a different directory or filename or something, this would be where that change would be made.